torch/csrc/jit/pickler.cpp

```diff
@@ -1,54 +1,81 @@
 #include <torch/csrc/jit/pickler.h>
+#include <string>
 
 namespace torch {
 namespace jit {
 
 using ::c10::IValue;
 
-PicklerClass getClass(const std::string& str) {
-  if (str == "TensorID") {
-    return PicklerClass::TENSOR;
-  } else if (str == "IntList") {
-    return PicklerClass::INTLIST;
-  }
-  AT_ERROR("Unknown class name for unpickler: ", str);
-}
+const static std::unordered_map<std::string, PicklerClass> name_to_class{
```

Comment:
What was wrong with how it was before? The other one is almost certainly faster.

Reply:
This is also missing an entry for LITERAL_TENSOR, right?

Reply:
It's a little more readable I think and this isn't really accessed that often to be a performance bottleneck or anything.

`LITERAL_TENSOR` wasn't necessary for this PR (it's only needed to Unpickle these tensors in C++, which isn't crucial since `torch.load` can read them in)